iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relocatable_ptr.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_RELOCATABLE_POINTER_RELOCATABLE_PTR_HPP
+#define IOX_HOOFS_RELOCATABLE_POINTER_RELOCATABLE_PTR_HPP
+
+#include <cstdint>
+#include <type_traits>
+
+namespace iox
+{
+namespace rp
+{
+/// @brief Smart pointer type that allows objects using it to be trivially copyable.
+///        This applies only if it points to memory owned by the object itself (i.e.
+///        i.e. not to memory outside of the object).
+///        This is useful to improve copy-efficiency and allow the types build with relocatable
+///        pointers only to be stored in shared memory.
+///        It is useable like a raw pointer of the corresponding type and can be implicily
+///        converted to one.
+///
+/// @tparam T the native type wrapped by the relocatable_ptr, i.e. relocatable_ptr<T>
+///         has native type T and corresponds to a raw pointer of type T*.
+///
+/// @note It is advisable to use relocatable_ptr only for storage (e.g. member variables),
+///       not to pass them around as function arguments or as return value.
+///       There should be no need for this, since as pass-around type
+///       regular pointers do the job just fine and do not incur
+///       the slight runtime overhead of a relocatable_ptr.
+///       There should be no memory overhead on 64 bit systems.
+template <typename T>
+class relocatable_ptr
+{
+  public:
+    /// @brief Construct from raw pointer.
+    relocatable_ptr(T* ptr = nullptr);
+    // {
+    //     m_offset = to_offset(ptr);
+    // }
+
+    /// @brief Construct from other relocatable pointer.
+    relocatable_ptr(const relocatable_ptr& other);
```

Comment:
This contradicts the class description since it states that the class is trivially copyable but this ctor make it non trivially copyable